### PR TITLE
Add tests to trigger space to switch mode

### PR DESCRIPTION
**Pull request by genti94**
- Add tests to trigger space to switch mode between structure and content

### CODE REVIEW

File: cms/tests/frontend/integration/spaceSwitch.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        .then(cms.removePage())
+        .then(cms.logout())
+        .run(done);
+});
+
+casper.test.begin('Switch mode with space', function (test) {
+    casper
+        .start(globals.editUrl)
+        // publishes page
+        .waitForSelector('.cms-btn-publish-active', function () {
```
Comment (vxsx) [inline cms/tests/frontend/integration/spaceSwitch.js]: no need to publish a page here

File: cms/tests/frontend/integration/spaceSwitch.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            test.assertExist('.cms-btn-switch-edit', 'published page');
+            // clicks on edit mode
+            this.click('.cms-btn-switch-edit');
+        })
+        // counts the element in structure mode
+        .waitUntilVisible('.cms-toolbar-expanded', function () {
+            test.assertElementCount(
+                '.cms-dragarea:nth-child(1) > .cms-draggables > .cms-draggable',
+                1,
+                'plugin is created'
```
Comment (vxsx) [inline cms/tests/frontend/integration/spaceSwitch.js]: no need to check for this as well

File: cms/tests/frontend/integration/spaceSwitch.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            );
+        })
+        .then(function () {
+            // triggers space
+            this.sendKeys('html', casper.page.event.key.Space);
+            // checks the name of the page which gets shown only on content mode
+            test.assertEquals(this.fetchText('.nav li a'),
```
Comment (vxsx) [inline cms/tests/frontend/integration/spaceSwitch.js]: this is plain wrong

File: cms/tests/frontend/integration/spaceSwitch.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            'spaceSwitch', 'switch via space worked');
+        })
+        .then(function () {
+            // triggers space again
+            this.sendKeys('html', casper.page.event.key.Space);
+        })
+            // counts the placeholder
+        .waitUntilVisible('.cms-toolbar-expanded', function () {
+            test.assertElementCount(
```
Comment (vxsx) [inline cms/tests/frontend/integration/spaceSwitch.js]: same here, wrong check

File: gulpfile.js [hunk 5]
```diff
@@ -202,7 +202,8 @@ gulp.task('tests:integration', function (done) {
         'history',
         'revertLive',
         'narrowScreen',
-        'clipboard'
+        'clipboard',
+        'spaceSwitch'
```
Comment (vxsx) [inline gulpfile.js]: i don't really think it deserves an extra file tbh, could've put it somewhere in existing structureboard tests

File: cms/tests/frontend/integration/spaceSwitch.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+// #############################################################################
+// Change Settings behaviour
```
Comment (vxsx) [inline cms/tests/frontend/integration/spaceSwitch.js]: copy paste

File: cms/tests/frontend/integration/createContent.js [hunk 7]
```diff
@@ -111,3 +111,28 @@ casper.test.begin('User Add Content', function (test) {
...
+        .then(cms.addPlugin({
+            type: 'TextPlugin',
+            content: {
+                id_body: 'sample text'
+            }
+        }))
+        .then(function () {
+            // triggers space
+            this.sendKeys('html', casper.page.event.key.Space);
+            // checks the name of the page which gets shown only on content mode
```
Comment (vxsx) [inline cms/tests/frontend/integration/createContent.js]: comment is wrong

**Comment by genti94** [comment]
@vxsx fixed

**Comment by vxsx** [comment]
:+1:

**Pull request merged**